src/react-components/home-root.js

```diff
@@ -35,9 +35,39 @@ class HomeRoot extends Component {
   componentDidMount() {
     this.loadEnvironments();
     this.setState({ dialogType: this.props.dialogType });
-    document.querySelector("#background-video").playbackRate = 0.75;
+    this.loadHomeVideo();
   }
 
+  loadHomeVideo = () => {
+    const videoEl = document.querySelector("#background-video");
+    function initVideo() {
+      videoEl.playbackRate = 0.75;
+      videoEl.play();
+      let inIframe = false;
+      try {
+        inIframe = !!window.frameElement || window.self !== window.top;
+      } catch (e) {
+        inIframe = false;
+      }
+      function toggleVideo() {
+        // Play the video if the window/tab is visible (or is within an `<iframe>`).
+        if (document.hasFocus() || inIframe) {
+          videoEl.play();
+        } else {
+          videoEl.pause();
+        }
+      }
+      document.addEventListener("visibilitychange", toggleVideo);
```

Comment:
Probably fine but just out of curiosity why do we need both of these? What cases do focus and blur catch that visiblitychange does not (and vis versa)?

Reply:
also, from the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API#Use_cases):

> Note: While `blur` and `focus` will tell you if the user switches windows, it doesn't necessarily mean it's hidden. Pages only become hidden when the user switches tabs or minimizes the browser window containing the tab.